test/python/transpiler/test_mappers.py

```diff
@@ -0,0 +1,280 @@
+
+        qr = QuantumRegister(3, 'q')               #
+        cr = ClassicalRegister(3, 'c')             # Set the circuit to test
+        circuit = QuantumCircuit(qr, cr,           # and don't forget to put a name
+                                 name='some_name') # (it will be used to save the pickle
+        circuit.h(qr[1])                           #
+        circuit.cx(qr[1], qr[2])                   #
+        circuit.measure(qr, cr)                    #
+
+        result = transpile(circuit, self.create_backend(), coupling_map=coupling_map,
+                           pass_manager=self.create_passmanager(coupling_map))
+        self.assertResult(result, circuit.name)
+```
+"""
+
+# pylint: disable=redefined-builtin,attribute-defined-outside-init
+
+import unittest
+import pickle
+
+from qiskit import ClassicalRegister, QuantumRegister, QuantumCircuit, BasicAer, compile
+from qiskit.transpiler import PassManager, transpile
+from qiskit.transpiler.passes import BasicSwap, LookaheadSwap, StochasticSwap
+from qiskit.mapper import CouplingMap, Layout
+
+from qiskit.test import QiskitTestCase
+
+
+class CommonUtilitiesMixin:
+    """Utilities for meta testing.
+
+    Subclasses should redefine the ``pass_class`` argument, with a Swap Mapper
+    class.
+
+    Note: This class assumes that the subclass is also inheriting from
+    ``QiskitTestCase``, and it uses ``QiskitTestCase`` methods directly.
+    """
+
+    regenerate_expected = False
+    seed = 42
+    additional_args = {}
+    pass_class = None
+
+    def create_passmanager(self, coupling_map, initial_layout=None):
+        """Returns a PassManager using self.pass_class(coupling_map, initial_layout)"""
+        passmanager = PassManager(
+            self.pass_class(CouplingMap(coupling_map),  # pylint: disable=not-callable
+                            **self.additional_args))
+        if initial_layout:
+            passmanager.property_set['layout'] = Layout(initial_layout)
+        return passmanager
+
+    def create_backend(self):
+        """Returns a Backend."""
+        return BasicAer.get_backend('qasm_simulator')
+
+    def generate_expected(self, transpiled_result, filename):
+        """Generates the expected result into a file.
+
+        Checks if transpiled_result matches self.counts by running in a backend
+        (self.create_backend()). That's saved in a pickle in filename.
+
+        Args:
+            transpiled_result (DAGCircuit): The DAGCircuit to compile and run.
+            filename (string): Where the pickle is saved.
+        """
+        sim_backend = self.create_backend()
+        qobj = compile(transpiled_result, sim_backend, seed=self.seed, shots=self.shots)
+        job = sim_backend.run(qobj)
+        self.assertDictAlmostEqual(self.counts, job.result().get_counts(), delta=self.delta)
+
+        with open(filename, "wb") as output_file:
+            pickle.dump(transpiled_result, output_file)
+
+    def assertResult(self, result, testname):
```

Comment:
Let’s do this method a little bit more abstract and instead of passing the `testname`, let’s pass the `circuit` so we can extract whatever string we want to use as name for the pickled file (actually, we can pass a hash to the textual representation of the circuit and avoid the name completely).

Reply:
But that would make the ground truth file name dependent on the textual representation of the circuit. I think the name of the circuit it can be literally choose to match a filename.

Reply:
Well, there are two things in this comment:
1. Passing `result` and `circuit` (not a `testname`).
2. Avoid using a name.

I'm not against using an explicitly provided name but I'm pro passing a circuit, and extract the name from it later.

Reply:
done in 6c7223c1